<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>710</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DuGu</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/images/microchip-solid.png</normaloff>:/images/microchip-solid.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <widget class="QGroupBox" name="gbPart">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>5</y>
      <width>331</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Part</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>19</y>
       <width>311</width>
       <height>81</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridPart">
      <item row="1" column="1">
       <widget class="QLineEdit" name="leHighFuse">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="inputMask">
         <string notr="true">HH</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="leLowFuse">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="inputMask">
         <string notr="true">HH</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="chbLowFuse">
        <property name="toolTip">
         <string>Check to write this fuse</string>
        </property>
        <property name="text">
         <string>Low fuse:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="chbHighFuse">
        <property name="toolTip">
         <string>Check to write this fuse</string>
        </property>
        <property name="text">
         <string>High fuse:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="leExtendedFuse">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="inputMask">
         <string notr="true">HH</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="chbExtendedFuse">
        <property name="toolTip">
         <string>Check to write this fuse</string>
        </property>
        <property name="text">
         <string>Extended fuse:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="chbSingleFuse">
        <property name="toolTip">
         <string>Check to write this fuse</string>
        </property>
        <property name="text">
         <string>Single fuse:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLineEdit" name="leSingleFuse">
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="inputMask">
         <string notr="true">HH</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="4">
       <widget class="QComboBox" name="cbPart">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbProgrammer">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>115</y>
      <width>331</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Programmer</string>
    </property>
    <widget class="QWidget" name="gridLayoutWidget_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>311</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridProgrammer">
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="leProgrammerPort">
        <property name="toolTip">
         <string>Enter port name to which programmer is connected</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QComboBox" name="cbProgrammer">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbFile">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>205</y>
      <width>331</width>
      <height>145</height>
     </rect>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>311</width>
       <height>111</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="layoutFile">
      <item>
       <widget class="QLineEdit" name="leFilePath">
        <property name="toolTip">
         <string>Set path to file which will be used as source of data or place to save readed data</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="bSelectFile">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Select file</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QGroupBox" name="gbOperationType">
          <property name="title">
           <string>Operation type</string>
          </property>
          <widget class="QWidget" name="verticalLayoutWidget_3">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>131</width>
             <height>31</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QComboBox" name="cbOperationType"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gbMemoryType">
          <property name="title">
           <string>Memory type</string>
          </property>
          <widget class="QWidget" name="verticalLayoutWidget_4">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>131</width>
             <height>31</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QComboBox" name="cbMemoryType"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbAdditionalOptions">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>331</width>
      <height>215</height>
     </rect>
    </property>
    <property name="title">
     <string>Additional options</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>19</y>
       <width>311</width>
       <height>181</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTableWidget" name="tbOptions">
        <property name="showDropIndicator" stdset="0">
         <bool>false</bool>
        </property>
        <property name="dragDropOverwriteMode">
         <bool>false</bool>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::NoSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="rowCount">
         <number>24</number>
        </property>
        <property name="columnCount">
         <number>2</number>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderCascadingSectionResizes">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>40</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderMinimumSectionSize">
         <number>30</number>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>30</number>
        </attribute>
        <attribute name="verticalHeaderStretchLastSection">
         <bool>false</bool>
        </attribute>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <column/>
        <column/>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="bStart">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>570</y>
      <width>331</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Start selected operation</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="pteOutput">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>10</y>
      <width>351</width>
      <height>556</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="undoRedoEnabled">
     <bool>false</bool>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
    <property name="placeholderText">
     <string>AVRDUDE output</string>
    </property>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>570</y>
      <width>351</width>
      <height>21</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>710</width>
     <height>22</height>
    </rect>
   </property>
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="locale">
     <locale language="English" country="UnitedStates"/>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOptions"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTargets">
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Targets</string>
    </property>
    <addaction name="actionAddCurrentAsNewTarget"/>
    <addaction name="actionRemoveSelectedTarget"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTargets"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionOptions">
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionAddCurrentAsNewTarget">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Add current as new target</string>
   </property>
  </action>
  <action name="actionRemoveSelectedTarget">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Remove selected target</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOptions</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showOptions()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAddCurrentAsNewTarget</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>addTarget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemoveSelectedTarget</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeTarget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bSelectFile</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setPathToDataFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bStart</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startOperation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showOptions()</slot>
  <slot>showAbout()</slot>
  <slot>addTarget()</slot>
  <slot>removeTarget()</slot>
  <slot>setPathToDataFile()</slot>
  <slot>startOperation()</slot>
 </slots>
</ui>
